api/utils.py

```diff
@@ -0,0 +1,89 @@
+import logging
+from functools import partial
+
+import jwt
+from django.conf import settings
+from django.core.exceptions import (
+    ImproperlyConfigured,
+    PermissionDenied,
+)
+
+logger = logging.getLogger(__name__)
+
+# ----------------------------------------------------------------------------------------------------------------------
+# JWT Authentication
+# https://github.com/Aalto-LeTech/django-lti-login/blob/jwt-api/django_lti_login/api_views.py
+
+
+def setting_in_bytes(name):
+    value = getattr(settings, name)
+    if isinstance(value, bytes):
+        return value
+    if isinstance(value, str):
+        return value.encode('utf-8')
+    raise ImproperlyConfigured(
+        "Value for settings.%s is not bytes or str."
+        % (name,))
+
+
+def prepare_decoder():
+    options = {'verify_' + k: True for k in ('iat', 'iss')}
+    options.update({'require_' + k: True for k in ('iat',)})
+    if hasattr(settings, 'JWT_ISSUER'):
+        options['issuer'] = settings.JWT_ISSUER
+
+    if hasattr(settings, 'JWT_PUBLIC_KEY'):
+        try:
+            from cryptography.hazmat.backends import default_backend
+            from cryptography.hazmat.primitives.serialization import load_pem_public_key
+        except ImportError as error:
+            raise ImproperlyConfigured(
+                "`mooc-grader api` requires `cryptography` when using settings.JWT_PUBLIC_KEY: %s"
+                % (error,))
+        pem = setting_in_bytes('JWT_PUBLIC_KEY')
+        try:
+            key = load_pem_public_key(pem, backend=default_backend())
+        except ValueError as error:
+            raise ImproperlyConfigured(
+                "Invalid public key in JWT_PUBLIC_KEY: %s"
+                % (error,))
+        return partial(jwt.decode,
+                       key=key,
+                       algorithms=settings.JWT_ALGORITHM,
+                       **options)
+    return None
+
+
+jwt_decode = prepare_decoder()
+
+
+def jwt_auth(request):
```

Comment:
You can refactor this to a decorator, and then it can be used like django's login_required (https://github.com/django/django/blob/master/django/contrib/auth/decorators.py#L38)